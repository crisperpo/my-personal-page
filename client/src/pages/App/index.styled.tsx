import styled from 'styled-components';

const HeaderContainer = styled.div`
    align-items: center;
    color: ${({theme}) => theme.text};
    display: flex;
    flex-direction: column;
    gap: 2rem;
    justify-content: center;
`;

export {
    HeaderContainer
};
