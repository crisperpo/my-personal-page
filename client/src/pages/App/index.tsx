import { useState, useEffect, useContext } from 'react';
import { ThemeProvider } from 'styled-components';

import fetchHomePageData from '../../services/fetchHomePageData';
import Title from '../../components/Title/';
import Headline from '../../components/Headline/';
import Social from '../../components/Social/';
import { HeaderContainer } from './index.styled';
import { AppContext } from '../../context/AppContext';
import theme from '../../lib/constants/theme';
import '../../styles/app.css';

const App = () => {
  const [mode, setMode] = useState('light')
  const [loading, setLoading] = useState(true);
  const appContext = useContext(AppContext);
  
  const { setPageData } = appContext || {};
  const currentTheme = theme[mode as keyof typeof theme] || theme.light;

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchHomePageData();
      if(setPageData) setPageData(data);
      setLoading(false);
    };
    fetchData();
  }, [setLoading, setPageData]);

  if (loading || !appContext) {
    return <div>...</div>;
  }

  return (
    <ThemeProvider theme={currentTheme}>
      <HeaderContainer>
         <Title />
        <Headline />
        <Social />
      </HeaderContainer>
    </ThemeProvider>
  );
};

export default App;
