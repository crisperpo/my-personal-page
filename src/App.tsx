import './styles/app.css';
import content from './lib/constants/content.json';

const App = () => {
  return (
    <>
      {content.title}
    </>
  )
};

export default App;
