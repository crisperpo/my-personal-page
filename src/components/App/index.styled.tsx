import styled from 'styled-components';

const HeaderContainer = styled.div`
    align-items: center;  
    display: flex;
    flex-direction: column;
    gap: 2rem;
    justify-content: center;
`;

export {
    HeaderContainer
};
