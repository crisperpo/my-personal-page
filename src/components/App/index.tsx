import Title from '@/components/Title/';
import Headline from '@/components/Headline/';
import Social from '@/components/Social/';
import { HeaderContainer } from './index.styled';
import '@/styles/app.css';

const App = () => (
  <>
    <HeaderContainer>
      <Title />
      <Headline />
      <Social />
    </HeaderContainer>
  </>
);

export default App;
