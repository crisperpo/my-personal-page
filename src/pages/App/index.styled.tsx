import styled from 'styled-components';

import colors from '../../lib/constants/colors';

const HeaderContainer = styled.div`
    align-items: center;
    color: ${colors.textBody};
    display: flex;
    flex-direction: column;
    gap: 2rem;
    justify-content: center;
`;

export {
    HeaderContainer
};
