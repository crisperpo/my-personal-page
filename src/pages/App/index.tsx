import { useState, useEffect, useContext } from 'react';

import fetchHomePageData from '../../services/fetchHomePageData';
import Title from '../../components/Title/';
import Headline from '../../components/Headline/';
import Social from '../../components/Social/';
import { HeaderContainer } from './index.styled';
import { AppContext } from '../../context/AppContext';
import '../../styles/app.css';

const App = () => {
  const [loading, setLoading] = useState(true);
  const appContext = useContext(AppContext);
  
  const { setPageData } = appContext || {};

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchHomePageData();
      if(setPageData) setPageData(data);
      setLoading(false);
    };
    fetchData();
  }, [setLoading, setPageData]);

  if (loading || !appContext) {
    return <div>...</div>;
  }

  return (
    <>
      <HeaderContainer>
        <Title />
        <Headline />
        <Social />
      </HeaderContainer>
    </>
  );
};

export default App;
