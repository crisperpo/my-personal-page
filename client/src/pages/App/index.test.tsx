import { describe, it, expect } from 'vitest';
import { act, render } from '@testing-library/react';
import { vi } from 'vitest';

import App from './index';
import mockContentfulData from '../../test/mockContentfulData.json';

vi.mock('../../services/fetchHomePageData', () => ({
    default: vi.fn(() => Promise.resolve(mockContentfulData)),
}));

describe('App Page', () => {
    it('renders correctly', async () => {
        await act(async () => {
            render(<App />);
        });
    });
    it('matches the snapshot', async () => {
        await act(async () => {
            const { asFragment } = render(<App />);
            expect(asFragment()).toMatchSnapshot();
        });
    });
});
