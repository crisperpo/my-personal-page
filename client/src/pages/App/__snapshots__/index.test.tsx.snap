// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Page > matches the snapshot 1`] = `<DocumentFragment />`;
